pkg/volume/host_path/host_path.go

```diff
@@ -216,6 +229,42 @@ func (r *hostPathRecycler) Recycle() error {
 	return volume.RecycleVolumeByWatchingPodUntilCompletion(pod, r.host.GetKubeClient())
 }
 
+// hostPathCreater implements a dynamic provisioning Creater for the HostPath plugin
+// This implementation is meant for testing only and only works in a single node cluster.
+type hostPathCreater struct {
+	host    volume.VolumeHost
+	options volume.VolumeOptions
+}
+
+// Create for hostPath simply creates a local /tmp/hostpath_pv/%s directory as a new PersistentVolume.
+// This Creater is meant for development and testing only and WILL NOT WORK in a multi-node cluster.
+func (r *hostPathCreater) Create() (*api.PersistentVolume, error) {
+	fullpath := fmt.Sprintf("/tmp/hostpath_pv/%s", util.NewUUID())
+	err := os.MkdirAll(fullpath, 0750)
+	if err != nil {
+		return nil, err
+	}
+
+	return &api.PersistentVolume{
+		ObjectMeta: api.ObjectMeta{
+			GenerateName: "pv-hostpath-",
+			Labels: map[string]string{
+				"createdby": "hostpath dynamic provisioner",
+			},
+		},
+		Spec: api.PersistentVolumeSpec{
+			Capacity: api.ResourceList{
+				api.ResourceName(api.ResourceStorage): resource.MustParse(fmt.Sprintf("%dMi", r.options.CapacityMB)),
```

Comment:
VolumeOptions is passed to the NewCreater(options) constructor func and plumbed through. 

This follows the NewBuilder(spec, options) pattern.